app/jobs/concerns/resource_request_jobs/iiif_job_behaviors.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ResourceRequestJobs
+  module IiifJobBehaviors
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def src_resource_for_digital_object(digital_object)
+        digital_object.asset_type == 'Image' ? digital_object.access_resource : digital_object.poster_resource
+      end
+
+      def eligible_object?(digital_object)
+        return false unless digital_object&.is_a?(::DigitalObject::Asset)
+        if digital_object.asset_type == 'Image'
+          digital_object.has_access_resource?
+        else
+          digital_object.has_poster_resource?
+        end
```

Comment:
This check applies to the IiifRegistrationJob, but not the IiifDeregistrationJob.  In addition, we also want to check if `digital_object.featured_thumbnail_region.present?` in a IiifRegistrationJob, but don't need to for a IiifDeregistrationJob.